delphi/app/pki_model.py

```diff
@@ -0,0 +1,106 @@
+from delphi.app.graph import run_optimization
+
+
+class PKI:
+    """
+    PKI is a class to represent the building PKI.
+
+    Attributes:
+        vertices: a dictionary of nodes and their enabled status
+        l2vertices: a dictionary of nodes and their enabled status
+        l3vertices: a dictionary of nodes and their enabled status
+        s1: the people count of room s1
+        s2: the people count of room s2
+        s3: the people count of room s3
+        desired_flow: the amount of people that need to exit pki
+        edges: a dictionary of edges and their weights
+        l2edges: a dictionary of edges and their weights
+        l3edges: a dictionary of edges and their weights
+    """
+
+    def __init__(self):
+        """
+        PKI constructor
+        """
+        self.vertices = {'t1': True, 't2': True, 's1': True, 's2': True, 's3': True, 'h1': True, 'h2': True, 'h5': True}
+        self.l2vertices = {'t3': True, 't4': True, 'h3': True, 'h4': True}
+        self.l3vertices = {'h6': True}
+        self.s1 = 25
+        self.s2 = 30
+        self.s3 = 20
+        self.setEdges()
+        self.desired_flow = self.s1 + self.s2 + self.s3
+
+    def toggle(self, node):
+        """
+        This method is designed to toggle the enabled status of the node passed in
+        :param node: the name of the ndoe to toggle
+        :return: the changed node and new status
+        """
+        if node in self.vertices:
+            for key in self.vertices:
+                if node == key:
+                    if self.vertices[key] is True:
+                        new = {key: False}
+                        self.vertices.update(new)
+                    else:
+                        new = {key: True}
+                        self.vertices.update(new)
+        elif node in self.l2vertices:
+            for key in self.l2vertices:
+                if node == key:
+                    if self.l2vertices[key] is True:
+                        new = {key: False}
+                        self.l2vertices.update(new)
+                    else:
+                        new = {key: True}
+                        self.l2vertices.update(new)
+        else:
+            for key in self.l3vertices:
+                if node == key:
+                    if self.l3vertices[key] is True:
+                        new = {key: False}
+                        self.l3vertices.update(new)
+                    else:
+                        new = {key: True}
+                        self.l3vertices.update(new)
```

Comment:
using the 'levels' dictionary i mentioned earlier would remove this duplicate code